modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -127,6 +128,19 @@ public function add_browse_sites_link() {
 		add_menu_page( __( 'Browse sites', 'jetpack' ), __( 'Browse sites', 'jetpack' ), 'read', 'https://wordpress.com/home', null, 'dashicons-arrow-left-alt2', 0 );
 	}
 
+	/**
+	 * Adds a link to the menu to create a new site.
+	 */
+	public function add_new_site_link() {
+		// Only show the menu on atomic or simple.
+		if ( ! jetpack_is_atomic_site() && ! $this->is_wpcom_site() ) {
+			return;
+		}
+
+		// Add the menu item.
+		add_menu_page( __( 'Add new site', 'jetpack' ), __( 'Add new site', 'jetpack' ), 'read', 'https://wordpress.com/start', null, 'dashicons-plus-alt', 98 );
```

Comment:
Potential improvement: some other places linking to `/start` add an additional GET param `?ref=…` to track where users came from into the signup. We'd need to come up with a param value for this exact place